src/osd/PGLog.h

```diff
@@ -753,6 +753,7 @@ struct PGLog : DoutPrefixProvider {
   void recover_got(hobject_t oid, eversion_t v, pg_info_t &info) {
     if (missing.is_missing(oid, v)) {
       missing.got(oid, v);
+      info.stats.stats.sum.num_objects_missing = missing.num_missing();
```

Comment:
For this, I think the on_local_recover() is the only caller that really matters, and publish_stats_to_osd() is already called a few lines down inside the if (is_primary()) blok.